pkg/apis/servicecatalog/types.go

```diff
@@ -439,3 +439,15 @@ const (
 const (
 	FinalizerServiceCatalog string = "kubernetes-incubator/service-catalog"
 )
+
+// Parameters is a union type for supporting different ways of representing parameters used for:
+// - Instance input parameters
+// - Binding output credentials
+type Parameters struct {
+	// Reference to a secret in the namespace.
```

Comment:
I think keeping parity with what we currently have is fine.  There is not a way, afaik, to only support json payloads in these fields.